windows/devices/uwb/UwbConnector.cxx

```diff
@@ -948,31 +1134,22 @@ UwbConnector::DeregisterEventCallback(std::weak_ptr<::uwb::RegisteredCallbackTok
         // treat it as a session callback
         auto sessionId = sessionCallback->SessionId;
 
-        auto node = m_sessionEventCallbacks.extract(sessionId);
-        if (node.empty()) {
-            return; // sessionId has no associated tokens
-        }
-
-        auto tokens = node.mapped();
-        auto tokenIt = std::find_if(std::cbegin(tokens), std::cend(tokens), [sessionCallback](const auto& token) {
-            return token.get() == sessionCallback.get();
-        });
-        if (tokenIt == std::cend(tokens)) {
-            return; // no associated token found, bail
-        }
-        tokens.erase(tokenIt);
-        m_sessionEventCallbacks.insert(std::move(node));
+        DeregisterSessionEventCallback<::uwb::OnSessionEndedToken>(tokenShared, m_onSessionEndedCallbacks)                               ? 0

```

Comment:
I don't quite get the use of the ternary operators here. Can't you just do:

`DeregisterSessionEventCallback<::uwb::OnSessionEndedToken>(tokenShared, m_onSessionEndedCallbacks);`
`DeregisterSessionEventCallback<::uwb::OnRangingStartedToken>(tokenShared, m_onRangingStartedCallbacks);`
`DeregisterSessionEventCallback<::uwb::OnRangingStoppedToken>(tokenShared, m_onRangingStoppedCallbacks);`
...

Reply:
The idea is to return early when it finds the correct class to cast to. I guess I could make it clearer by using if statements, but I think ternary statements are more concise.

Reply:
Oh I see. Well in that case, I think you could just use `||` to short-circuit the code:

`DeregisterSessionEventCallback<::uwb::OnSessionEndedToken>(tokenShared, m_onSessionEndedCallbacks) || DeregisterSessionEventCallback<::uwb::OnRangingStartedToken>(tokenShared, m_onRangingStartedCallbacks) || ...`